gateway/loaders/zookeeper/utils.go

```diff
@@ -17,6 +22,55 @@ type ZookeeperClient struct {
 	conn *zk.Conn
 }
 
+func (z *ZookeeperTargetLoader) refreshZookeeperConfig(log zerolog.Logger, targetChan chan<- *connections.TargetConnectionControl, config *configuration.GatewayConfig) error {
+	if exists, _, _ := z.zkClient.conn.Exists(GlobalCertPath); exists {
+		tlsData, err := z.zkClient.getNodeValue(GlobalCertPath)
+		if err != nil {
+			return err
+		}
+		certConf := &CertificateConfig{}
+		if err := yaml.Unmarshal(tlsData, certConf); err != nil {
+			return err
+		}
+
+		tlsConfig := &tls.Config{
+			Renegotiation: tls.RenegotiateNever,
+		}
+
+		if len(certConf.ClientCA) == 0 {
+			tlsConfig.InsecureSkipVerify = true
+		} else {
+			certPool := x509.NewCertPool()
+			if ok := certPool.AppendCertsFromPEM([]byte(certConf.ClientCA)); !ok {
+				return errors.New("invalid ClientCA certificate")
+			}
+			tlsConfig.RootCAs = certPool
+		}
+
+		if len(certConf.ClientCert) > 0 && len(certConf.ClientKey) > 0 {
+			cert, err := tls.X509KeyPair([]byte(certConf.ClientCert), []byte(certConf.ClientKey))
+			if err != nil {
+				return err
+			}
+			tlsConfig.Certificates = []tls.Certificate{cert}
+		} else {
+			return errors.New("client cert or key has length 0, discarding TLS config")
+		}
+
+		config.ClientTLSConfig = tlsConfig
+
+		config.Log.Info().Msg("Gateway TLS Config set. Using TLS client authentication")
+
+		controlMsg := new(connections.TargetConnectionControl)
+		controlMsg.ReconnectAll = true
+		targetChan <- controlMsg
+	} else {
+		config.Log.Info().Msg("No TLS Certificates found")
```

Comment:
nit: let's log something like "No global config found", we may eventually use it for settings other than TLS certificates.

Reply:
the check is performed for the TLS config path, specifically /certificates/global

Reply:
ohh, my bad, I think I misread that